kiwi/shareddata.h

```diff
@@ -7,6 +7,16 @@
 |----------------------------------------------------------------------------*/
 #pragma once
 
+/*
+Implementation note
+===================
+SharedDataPtr/SharedData offers the same basic functionality as std::shared_ptr,
+but without using atomic counters.
+Since kiwi operates within a single thread context, atomic counters are not necessary,
+while they have a CPU cost.
+Therefore the use of SharedDataPtr/SharedData is preferred over std::shared_ptr.
```

Comment:
```suggestion
```

This is implied given that there's a comment justifying their use in the first place. It can be removed without losing any information.

Reply:
Being extra explicit here does not cost us much so I am going to keep this.